core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -372,6 +377,18 @@
 
     private static final boolean VALIDATE_HEADERS = getBoolean("validateHeaders", true);
 
+    private static final Set<TransientServiceOption> TRANSIENT_SERVICE_OPTIONS =
+            Streams.stream(CSV_SPLITTER.split(getNormalized("transientServiceOptions", "", val -> {
+                try {
+                    Streams.stream(CSV_SPLITTER.split(val))
+                           .forEach(feature -> TransientServiceOption.valueOf(Ascii.toUpperCase(feature)));
+                    return true;
+                } catch (Exception e) {
+                    return false;
+                }
+            }))).map(feature -> TransientServiceOption.valueOf(Ascii.toUpperCase(feature)))
+                   .collect(toImmutableSet());
```

Comment:
Could we make this an immutable enum set? Otherwise, `TransientServiceOptionsBuilder.build()` will sometimes return an immutable enum set and sometimes not.

Reply:
That's a good suggestion. Fixed 😉